.circleci/config.yml

```diff
@@ -233,3 +269,12 @@ workflows:
           filters:
             branches:
               only: master
+      - pypi-deploy:
+          requires:
+            - install
+            - mypy
+            - pytests
+            - windows-pytests
+          filters:
+            tags:
+              only: /(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(\.post(0|[1-9][0-9]*))?/
```

Comment:
It is possible that you need to add the `filters/tags` to all CI jobs as well. IIRC CircleCI is not smart enough to run dependencies (for instance `install`) when a child job (`pypi-deploy`) has to run

Reply:
Oh? That would be a pain :s 
I'll try to find a way without this :s but I suppose you're right :(

Reply:
For now at least, I could get it to not run when tag is missing by adding
```
branches:
  ignore: /.*/
```
which does not make sense to me.
What your saying is that if I push a tag, only this part may run? That seems weird

Reply:
Oh the branch ignore is because by default all branches are included, and there's a OR condition so any branch OR a tagged branch would trigger it if we dont ignore all branches... makes more sense